engine/step/step.go

```diff
@@ -170,20 +207,38 @@ func (st *Step) generateExecution(action executor.Executor, baseConfig map[strin
 
 	for { // until we break because no more functions
 
-		if len(action.BaseOutput) > 0 {
-			base, err := rawResolveObject(values, action.BaseOutput, st.Item, st.Name)
+		if action.Output != nil && action.Output.Strategy != executor.OutputStrategynone {
+			var skipAdd bool
+
+			if action.Output.Strategy == executor.OutputStrategymerge {
+				// XXX: Backward compatibility
```

Comment:
imo it shouldn't: we need to resolve the values, and in the future we won't have any RawMessage here, it will be the type unmarshalled, so we need to handle the case